packages/make-styles/src/runtime/resolveStyleRules.ts

```diff
@@ -87,16 +88,31 @@ export function resolveStyleRules(
 
         resolveStyleRules(value, unstable_cssPriority, pseudo, media, combinedSupportQuery, result);
       } else if (property === 'animationName') {
-        // TODO: support RTL!
-        const keyframe = compileKeyframeRule(value);
-        const animationName = HASH_PREFIX + hashString(keyframe);
+        const animationNames = Array.isArray(value) ? value : [value];
+        let keyframeCSS = '';
+        let keyframeRtlCSS = '';
+        const names = [];
+        const namesRtl = [];
+        for (const val of animationNames) {
+          const keyframe = compileKeyframeRule(val);
+          const name = HASH_PREFIX + hashString(keyframe);
+          keyframeCSS += compileKeyframes(name, keyframe);
+          names.push(name);
 
-        // TODO call Stylis for prefixing
-        const keyframeCSS = `@keyframes ${animationName}{${keyframe}}`;
+          const rtlKeyframe = compileKeyframeRule(convert(val));
+          if (keyframe !== rtlKeyframe) {
+            const nameRtl = RTL_PREFIX + name;
+            keyframeRtlCSS += compileKeyframes(nameRtl, rtlKeyframe);
+            namesRtl.push(nameRtl);
+          } else {
+            namesRtl.push(name);
+          }
+        }
 
-        result[animationName] = [animationName, keyframeCSS /* rtlCSS */];
-
-        resolveStyleRules({ animationName }, unstable_cssPriority, pseudo, media, support, result);
+        const animationName = names.join(' ');
+        const animationNameRtl = namesRtl.join(' ');
+        result[animationName] = [undefined, keyframeCSS, keyframeRtlCSS || undefined];
```

Comment:
Here I changed `[0]` from `animationName` to `undefined` - animationName should not be passed as className